features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Apis/Details/Comments/Comment.jsx

```diff
@@ -327,11 +335,56 @@ class Comment extends React.Component {
                                             allComments={allComments}
                                         />
                                     )} */}
+                                        {comment.replies && comment.replies.list.slice(0).map((reply, index) => (
+                                            <>
+                                                <Box ml={8}>
+                                                    {index !== 0 && <Divider className={classes.divider} />}
+                                                    <Grid container spacing={1} className={classes.root}>
+                                                        <Grid item>
+                                                            <Icon className={classes.commentIcon}>account_circle</Icon>
+                                                        </Grid>
+                                                        <Grid item xs zeroMinWidth>
+                                                            <Typography noWrap className={classes.commentText}>
+                                                                {(reply.commenterInfo && reply.commenterInfo.fullName)
+                                                                    ? reply.commenterInfo.fullName : reply.createdBy}
+                                                            </Typography>
+                                                            <Typography noWrap className={classes.commentText} variant='caption'>
+                                                                {moment(reply.createdTime).fromNow()}
+                                                            </Typography>
+
+                                                            {index !== editIndex && (
+                                                                <Typography className={classes.commentText}>
+                                                                    {reply.content}</Typography>
+                                                            )}
+
+                                                            {index === editIndex && (
```

Comment:
I think using the array index to identify which comment is being edited is a bit tricky because if someone adds or removes a reply/comment while editing, these indexes will get changed, Shall we use comment id instead, because that will be unique where ever the comment is placed in the array?

Reply:
And also I think this is not scalable when we have more nested commenting levels